tangelo/algorithms/classical/mp2_solver.py

```diff
@@ -49,14 +50,25 @@ def __init__(self, molecule):
         self.frozen = molecule.frozen_mos
         self.uhf = molecule.uhf
 
+        if self.spin != 0 or self.uhf:
```

Comment:
I need a comment here. What is this code block doing ? What is it providing that wasn't available before ?